crates/seal-committee/src/utils.rs

```diff
@@ -0,0 +1,106 @@
+// Copyright (c), Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Utility helper functions for working with Seal protocol types.
+
+use crate::seal_move_types::SealCommittee;
+use anyhow::Result;
+use fastcrypto::encoding::{Encoding, Hex};
+use fastcrypto::groups::bls12381::{G2Element, Scalar as G2Scalar};
+use fastcrypto_tbls::ecies_v1::{PrivateKey, PublicKey};
+use serde::{Deserialize, Serialize};
+
+/// Build a mapping from new committee party IDs to old committee party IDs.
+/// This is used for key rotation to identify which members are continuing from the old committee.
+pub fn build_new_to_old_map(
+    new_committee: &SealCommittee,
+    old_committee: &SealCommittee,
+) -> std::collections::HashMap<u16, u16> {
+    let mut new_to_old_map = std::collections::HashMap::new();
+    new_committee
+        .members
+        .iter()
+        .enumerate()
+        .for_each(|(party_id, address)| {
+            if let Ok(old_party_id) = old_committee.get_party_id(address) {
+                new_to_old_map.insert(party_id as u16, old_party_id);
+            }
+        });
+    new_to_old_map
+}
+
+/// Keys file structure for JSON/
+#[derive(Serialize, Deserialize)]
+pub struct KeysFile {
+    #[serde(
```

Comment:
since we control the format, why not use the existing ser/deser of all the types here (or add them where needed)?

Reply:
ah i remember the problem. the move byte deserleriazer is using x\"0x... not a convention hex. its not compatible to use it for serde_json for file ser/de for enc_sk, enc_pk, signing_sk, signing_pk. its the most ergonomic if we just use hex for json. the x0x string is whats returned from grpc, so it had to be handled with its own deserializer (just for enc_pk and signing_pk).

Reply:
KeysFile only uses fastcrypto types, how is that related to move byte deserleriazer?

Reply:
```
base64_to_hex() {
    echo -n "$1" | base64 -d 2>/dev/null | xxd -p -c 256 | sed 's/^/0x/'
}

ENC_PK_B64=$(jq -r '.enc_pk' .dkg.key)
SIGNING_PK_B64=$(jq -r '.signing_pk' .dkg.key)

export DKG_ENC_PK=$(base64_to_hex "$ENC_PK_B64")
export DKG_SIGNING_PK=$(base64_to_hex "$SIGNING_PK_B64")
```